lib/utils/net.go

```diff
@@ -35,3 +37,23 @@ func ClientIPFromConn(conn net.Conn) (string, error) {
 
 	return clientIP, nil
 }
+
+// InferProxyPublicAddr tries to extract a proxyDNSName from a given fqdn and the list of proxy public addrs.
+// If a proxyDNS name is not found, the default param will be returned.
+func InferProxyPublicAddr(fqdn string, proxyDNSNames []string, defaultPublicAddr string) string {
+	if fqdn == "" || defaultPublicAddr == "" {
+		return defaultPublicAddr
+	}
```

Comment:
Two questions: 
1. What is the source of `defaultPublicAddr`? Does it come from app config spec or is calculated from some other values? 
2. What are the implication of returning an empty string? Should we return an error here if the value is empty?

Reply:
the source is at the callsite. its up to the caller to pass in a default value, and if they pass in an empty string, it would be because they want to default to empty if the proxyDNSName cannot be inferred (i dont have a use case for it, but i dont think an error should be called here)

however, i wont die on this hill. tbh, the signature without any error is easier to deal with where this is called but, thats my only excuse

Reply:
i think that argument `defaultAddr` should be renamed to `defaultProxyPublicAddr`, as its the default addr for the proxy to use, rather than anything that is derived from the request host. it is NOT the default addr for the app

Reply:
> Alternatively, if we are sure that the empty return value is handeld at every call site, then maybe just explaining that in the comment would be ok too.

i can update the comment. however, since the caller is the one passing in the default argument, i would deem that as "up to the caller" to decide if the default should be empty or not, no? it should not be a surprise if an empty default is returned, if thats what the caller input as the default.

regardless, i will make it super clear, because thats always a good idea

Reply:
Actually, i can just remove the `default` argument, and return empty string if no proxyDNSNames are passed in, or just `proxyDNSNames[0]` (this is how the apps public addr is built now, just on the first dns name, no other logic).

i think that'd make it simpler, and ill add godoc to point that out